learning/step_by_step/scripting.rst

```diff
@@ -7,75 +7,86 @@ Introduction
 ------------
 
 Much has been said about tools that allow users to create video games
-without programming. It's been a dream for many independent developers
-to create games without learning how to code. This need has been around
-for a long time, even inside companies, where game designers wish to
-have more control of the game flow.
-
-Many products have been shipped promising a no-programming environment,
-but the result is often incomplete, too complex or inefficient compared
-to traditional code. As a result, programming is here to stay for a long
-time. In fact, the general direction in game engines has been to add
-tools that try to reduce the amount of code that needs to be written for
-specific tasks, to speed up development.
-
-In that sense, Godot has taken some useful design decisions towards that
-goal. The first and most important is the scene system. The aim of it is
-not obvious at first, but works well later on. That is, to relieve
-programmers from the responsibility of architecting code.
-
-When designing games using the scene system, the whole project is
-fragmented into *complementary* scenes (not individual ones). Scenes
-complement each other, instead of being separate. There will be plenty
-of examples about this later on, but it's very important to remember it.
-
-For those with a good amount of programming expertise, this means a
-different design pattern to MVC. Godot promises efficiency at the
-expense of dropping the MVC habits, which are replaced by the *scenes as
-a complement* pattern.
-
-Godot also uses the `extend <http://c2.com/cgi/wiki?EmbedVsExtend>`__
-pattern for scripting, meaning that scripts extend from all the
-available engine classes.
+without programming. This appeals to many people who would love to be able to
+make a game without having to learn how to code. Furthermore, in many game
+companies, developers write custom tools that give their game designers more
+control over their game's flow. It's tempting, therefore, to imagine that
+someone could could write the ultimate tool that just works for any game type.
+
+Many products have been shipped promising such a no-programming environment,
+but the results often fall short of expectations. The final projects they
+produce regularly end up being too complex or require solutions that are too
+inefficient compared to what could have been accomplished with traditional code.
+
+In contrast, we believe that programming is a critical part of game development
```

Comment:
Thanks. I've reverted to something closer to the original sentence.

The problem I was trying to solve is that the original sentence was

"As a result, programming is here to stay for a long time."

This is stated as fact, but we can't know 100% for sure. Something like "We feel" or "We think" seems required here.

Let me know what you think of this rewrite: " As a result, we think that programming is here to stay." Does that feel less marketing to you?